plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -140,20 +360,31 @@ class bp_connection_manager {
                fc_dlog(self()->get_logger(), "pending producer schedule switches from version ${old} to ${new}",
                        ("old", pending_schedule_version)("new", schedule.version));
 
-               auto pending_connections = configured_bp_accounts(config, schedule.producers);
+               auto pending_connections = active_bp_accounts(schedule.producers);
 
                fc_dlog(self()->get_logger(), "pending_connections: ${c}", ("c", to_string(pending_connections)));
-               for (const auto& i : pending_connections) {
-                  self()->connections.resolve_and_connect(config.bp_peer_addresses[i], self()->get_first_p2p_address() );
+
+               fc::lock_guard g(gossip_bps.mtx);
+               auto& prod_idx = gossip_bps.index.get<by_producer>();
+               for (const auto& account : pending_connections) {
+                  if (auto i = config.bp_peer_addresses.find(account); i != config.bp_peer_addresses.end()) {
+                     fc_dlog(self()->get_logger(), "connect to manual bp peer ${p}", ("p", i->second));
+                     self()->connections.resolve_and_connect(i->second, self()->get_first_p2p_address() );
+                  }
+                  auto r = prod_idx.equal_range(account);
+                  for (auto i = r.first; i != r.second; ++i) {
+                     fc_dlog(self()->get_logger(), "connect to gossip bp peer ${p}", ("p", i->server_address));
+                     self()->connections.resolve_and_connect(i->server_address, self()->get_first_p2p_address() );
```

Comment:
should we skip this connection attempt (and log message) if we already connected to the same `i->server_address` on line 372?

Reply:
`resolve_and_connect` does skip if already connected. It doesn't log it. Could add a log statement, might be useful.

Reply:
Oh, ok. Those will now be followed up with what happens.